src/main/java/frc/robot/controllers/DriverController.java

```diff
@@ -109,6 +116,11 @@ public void setRumble(double rumble) {
   }
 
   public Command getRumbleCommand(double rumble, double seconds) {
-    return Commands.runEnd(()->{setRumble(rumble);}, ()->{setRumble(0);}).withTimeout(seconds);
+    return Commands.runEnd(()->setRumble(rumble), ()->setRumble(0)).withTimeout(seconds);
+  }
+
+  public Command getRumbleCommand(double rumble, double seconds, int pulses) {
+    final int pulseCounter[] = {0};
```

Comment:
It's final for a bit of a weird reason - lambdas/runnables in java don't let you reference non-final variables from outside of their scope (likely has to do with them being stack-allocated and objects existing on the JVM equivalent to a heap). Not precisely sure how to move that outside of its current state short of rewriting the whole thing. Uppercasing it seems a bit weird since even though it's final, it's still being edited (just the underlying object, not the variable), if that makes sense.